docker_test.go

```diff
@@ -1384,20 +1384,25 @@ func TestContainerNonExistentImage(t *testing.T) {
 }
 
 func TestContainerCustomPlatformImage(t *testing.T) {
-	t.Run("Use a custom image platform", func(t *testing.T) {
+	t.Run("Use a non-existent custom image platform", func(t *testing.T) {
+		nonExistentPlatform := "windows/arm12"
 		c, err := GenericContainer(context.Background(), GenericContainerRequest{
 			ContainerRequest: ContainerRequest{
 				Image:         "redis:latest",
 				SkipReaper:    true,
-				ImagePlatform: "windows/arm12",
+				ImagePlatform: nonExistentPlatform,
 			},
-			Started: false,
+			Started: true,
 		})
 		if c != nil {
 			defer c.Terminate(context.Background())
 		}
 		if err == nil {
-			t.Fatalf("Expected non-nil error")
+			t.Fatalf("Expected an error containing failed to create container: Error response from daemon")
+		}
+		expectedErrorMsgSubstring := "image with reference redis:latest was found but does not match the specified platform"
```

Comment:
Mmm, not sure about this new verification. Will that message change between Docker versions? If you see it convenient, I'm OK with it too, but I'd like to make sure we are not adding flakiness because of specific message formats. 

```go
if err == nil {
	t.Fatalf("Expected non-nil error with a non-existent platform: %s", nonExistentPlatform)
}
```

Reply:
Sounds like a good idea. done.